### PR TITLE
[cloud-provider-zvirt] Fix empty private IP addresses

### PR DESCRIPTION
**Pull request by ilia601**
## Description
Added a condition to use external addresses if internal ones are missing

## Why do we need it, and what problem does it solve?
So that modules such as monitoring-ping work correctly. [Monitoring-ping pods will crash if there are no targets to ping](https://github.com/deckhouse/deckhouse/blob/main/modules/340-monitoring-ping/images/monitoring-ping/monitoring-ping.py#L116)

## What is the expected result?
zvirt nodes have internal addresses. monitoring-ping pods work as expected without crashes

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: monitoring-ping
type: fix
summary: Fix monitoring-ping pods crashing
impact_level: default
```

### CODE REVIEW

**Comment by ilia601** [comment]
```
Traceback (most recent call last):
  File "/src/monitoring-ping.py", line 181, in <module>
    computed = compute_results(out)
               ^^^^^^^^^^^^^^^^^^^^
  File "<decorator-gen-1>", line 2, in compute_results
  File "/opt/python-static/lib/python3.11/site-packages/prometheus_client/context_managers.py", line 23, in wrapped
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/src/monitoring-ping.py", line 116, in compute_results
    raise ValueError("regex match\"{}\" found nothing in fping output \"{}\"".format(FPING_REGEX, results))
ValueError: regex match"re.compile('^(\\S*)\\s*: (.*)$', re.MULTILINE)" found nothing in fping output ""
```

Screenshot 2024-08-28 at 21 07 00

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @ilia601. I've started the workflow for label 'e2e/run/openstack'...

**Comment by ilia601** [comment]
/e2e/abort openstack;Standard;containerd;1.27 10602482297 10602482297-1-con-1-27 /sys/deckhouse-oss/install:pr9533 redos@5.159.103.12 - for abort failed cluster

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @ilia601. I've started the workflow for label 'e2e/run/openstack'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @ilia601. I've started the workflow for label 'e2e/run/openstack'...

**Comment by ilia601** [comment]
Screenshot 2024-09-03 at 11 51 16
Screenshot 2024-09-03 at 11 52 13
Screenshot 2024-09-03 at 11 52 53

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @ilia601. I've started the workflow for label 'e2e/run/openstack'...

File: ee/modules/030-cloud-provider-zvirt/images/cloud-controller-manager/src/pkg/cloudprovider/zvirt/instances.go [hunk 1]
```diff
@@ -125,6 +125,10 @@ func (zc *Cloud) extractNodeAddressesFromVM(ctx context.Context, vm ovirtclient.
 		})
 	}
 
+	if len(localIP) < 1 {
+		localIP = externalIP
+	}
```
Comment (aleksey-su) [inline ee/modules/030-cloud-provider-zvirt/images/cloud-controller-manager/src/pkg/cloudprovider/zvirt/instances.go:130]:
```suggestion
    // If there are no local IP addresses, replace them with external ones
	if len(localIP) == 0 {
		localIP = externalIP
	}
```

Comment (aleksey-su) [reply ee/modules/030-cloud-provider-zvirt/images/cloud-controller-manager/src/pkg/cloudprovider/zvirt/instances.go:130]: You should specify an explicit check condition to make it easier to read the code. And leave comments if you add non-standard behavior

**Review by aleksey-su — CHANGES_REQUESTED** [review]

**Review by aleksey-su — APPROVED** [review]

**Comment by alexey-gavrilov-flant** [comment]
/e2e/abort openstack;Standard;containerd;1.27 10602482297 10602482297-1-con-1-27 /sys/deckhouse-oss/install:pr9533 redos@5.159.103.12

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @alexey-gavrilov-flant. I've recognized your '/e2e/abort' command and started the workflow...

**Pull request merged**